<launch>

  <!-- ******************** ARGS ******************** -->
  <arg name="image_process" default="true"/>
  <arg name="repo_path" default="/home/chris/Documents/jackal_fyp"/>
  <arg name="use_webcam" default="true"/>

  <!-- ******************** USB WEBCAM  ******************** -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>

  <!-- ******************** IMAGE PROCESSING ******************** -->
  <node if="$(arg image_process)" name="image_processor" pkg="development" type="img_proc.py" output="screen">
    <param name="root" value="$(arg repo_path)"/>
    <param name="webcam" value="$(arg use_webcam)"/>
  </node>

</launch>
